LFS-728: Make logo, app name, and side bar background image configurable

OrdiNeu opened the pull request:
Running the executable in CardiacDB mode (`-Dsling.run.modes=nolfs,cardiac_rehab,dev`) now changes the logo and background image, and app name.

OrdiNeu committed:
LFS-728: Make logo, app name, and side bar background image configurable

Switch the logo depending on the current run mode

OrdiNeu committed:
LFS-728: Make logo, app name, and side bar background image configurable

Move the background to a place where it can be easily overwritten as
well

OrdiNeu committed:
LFS-728: Make logo, app name, and side bar background image configurable

Move the title to a meta property and load it via HTL

OrdiNeu committed:
LFS-728: Make logo, app name, and side bar background image configurable

Substitute the title elsewhere in the application

marta- commented on modules/homepage/src/main/frontend/src/themePage/index.jsx:
```diff
@@ -110,8 +110,8 @@ class Main extends React.Component {
               contentOffset={this.state.contentOffset}
-              logoText={"LFS Data Core"}
-              logoImage={"/libs/lfs/resources/lfs-logo-tmp-cyan.png"}
+              logoText={this.state.title + " Data Core"}
```
LFS Data Core should be the name of the app in LFS runmode

marta- replied:
More generally:
* the app name / first segment of the page title for the 3 run modes are : "LFS Data Core", "Cards 4 Kids", "Cards 4 CaRe"
* "data core" is not part of the general UI, it only belongs to LFS
* the sidebar should only display the logo, not the appname/title, centered horizontally in the sidebar
* we need an updated LFS Data Core that includes the text and is wider to compensate for the removal of the text "LFS Data Core" from the side bar

OrdiNeu committed:
LFS-728: Make logo, app name, and side bar background image configurable

Add cardiac runmode, remove the app title from the sidebar and center
the image.

sashaandjic commented:
Logo appears to be too small
![image](https://user-images.githubusercontent.com/23243244/101213990-439ff100-3649-11eb-995d-11bbc3cbf1be.png)

marta- commented:
> Logo appears to be too small

Also it's not centered

sashaandjic commented:
Everything else - perfect! Here are few screenshots:
![image](https://user-images.githubusercontent.com/23243244/101214372-ece6e700-3649-11eb-84db-acea823d007f.png)
![image](https://user-images.githubusercontent.com/23243244/101214382-f2443180-3649-11eb-9781-de602814694d.png)
![image](https://user-images.githubusercontent.com/23243244/101214394-f7a17c00-3649-11eb-9a31-e6a1100fcffa.png)
![image](https://user-images.githubusercontent.com/23243244/101214409-fc663000-3649-11eb-911a-9e035ecf7c27.png)
![image](https://user-images.githubusercontent.com/23243244/101214422-01c37a80-364a-11eb-914b-755847e633c2.png)

OrdiNeu committed:
LFS-728: Make logo, app name, and side bar background image configurable

Re-center the image

OrdiNeu committed:
LFS-728: Make logo, app name, and side bar background image configurable

Update the Cards4Kids logo

sashaandjic commented:
Almost perfect. Logo for cardiac mode needs to be centered and a bit bigger.
![image](https://user-images.githubusercontent.com/23243244/101221546-e6f70300-3655-11eb-9497-bf3e0cc6467a.png)

OrdiNeu commented:
I tried my best to cut down the blank space at the top&bottom of that logo, but can't make it any larger without distorting the appearance of the other logos.

marta- commented on modules/homepage/src/main/frontend/src/themePage/Sidebar/sidebarStyle.jsx:
```diff
@@ -88,17 +87,18 @@ import {
-      width: "30px",
+      width: "100%",
       display: "inline-block",
       maxHeight: "30px",
```
This should be removed to work well with taller horizontal logos.
```suggestion
```

marta- commented on modules/homepage/src/main/frontend/src/themePage/Sidebar/sidebarStyle.jsx:
```diff
@@ -88,17 +87,18 @@ import {
     img: {
-      width: "32px",
+      maxWidth: "90%",
+      maxHeight: "30px",
```
This should either be increased or removed to accommodate taller horizontal logos like the one for care4kids. 80px seems to work fine for our current logo shapes.
```suggestion
      maxHeight: "80px",
```

marta- commented on modules/homepage/src/main/frontend/src/themePage/Sidebar/sidebarStyle.jsx:
```diff
@@ -88,17 +87,18 @@ import {
     },
     img: {
-      width: "32px",
+      maxWidth: "90%",
```
This gives the logo some space left and right after we relax the `maxHeight`.
```suggestion
      maxWidth: "70%",
```

marta- committed:
LFS-728: Make logo, app name, and side bar background image configurable

Improve logo display for different logo shapes

sashaandjic reviewed (approved):
tested
![image](https://user-images.githubusercontent.com/23243244/101231288-2e8e8680-3678-11eb-9f36-f3575ca4ced3.png)

marta- reviewed (approved):
Looks good to me
